### PR TITLE
*: deny unknown fields when deserializing config TOML

### PR DESCRIPTION
**Pull request by kennytm**
## What have you changed? (mandatory)

Add [`#[serde(deny_unknown_fields)]`](https://serde.rs/container-attrs.html#deny_unknown_fields) to all config structures, which will cause deserialization to fail if the source contains an unrecognized field. The rationale to catch typo or misplaced settings.

## What are the type of the changes? (mandatory)

- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

```sh
cargo test --test integrations -- test_error_on_unrecognized_config
```

## Does this PR affect documentation (docs) or release note? (mandatory)

Release note.

## Does this PR affect tidb-ansible update? (mandatory)

Hopefully no 🙃.

## Refer to a related PR or issue link (optional)

pingcap/tidb#9855

### CODE REVIEW

**Comment by breeswish** [comment]
Does this PR need to be cherry picked to 3.0?

**Comment by kennytm** [comment]
@breeswish We could. WDYT

**Comment by breeswish** [comment]
@BusyJay @overvenus  Do you have some ideas?

File: src/config.rs [hunk 1]
```diff
@@ -75,6 +75,7 @@ fn memory_mb_for_cf(is_raft_db: bool, cf: &str) -> usize {
 
 #[derive(Clone, Serialize, Deserialize, PartialEq, Debug)]
 #[serde(default)]
+#[serde(deny_unknown_fields)]
```
Comment (overvenus) [inline src/config.rs]: Seems we can no longer downgrade if there is a new config exist?

Comment (kennytm) [reply src/config.rs]: You'd need to comment out that config manually. The error message will tell you the name of the offending field.

Comment (siddontang) [reply src/config.rs]: seem TiDB has the same problem now. is it ok?
@shenli

Comment (shenli) [reply src/config.rs]: yes, I think so.

**Comment by breeswish** [comment]
Need cherry pick? @siddontang @BusyJay

**Pull request merged**